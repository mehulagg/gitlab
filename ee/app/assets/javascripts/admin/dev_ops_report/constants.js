import { s__, __ } from '~/locale';

export const MAX_REQUEST_COUNT = 10;

export const DEVOPS_ADOPTION_SEGMENT_MODAL_ID = 'devopsSegmentModal';

export const DEVOPS_ADOPTION_SEGMENT_DELETE_MODAL_ID = 'devopsSegmentDeleteModal';

export const DATE_TIME_FORMAT = 'yyyy-mm-dd HH:MM';

export const DEVOPS_ADOPTION_ERROR_KEYS = {
  groups: 'groupsError',
  segments: 'segmentsError',
};

export const DEVOPS_ADOPTION_STRINGS = {
  app: {
    [DEVOPS_ADOPTION_ERROR_KEYS.groups]: s__(
      'DevopsAdoption|There was an error fetching Groups. Please refresh the page to try again.',
    ),
    [DEVOPS_ADOPTION_ERROR_KEYS.segments]: s__(
      'DevopsAdoption|There was an error fetching Segments. Please refresh the page to try again.',
    ),
    tableHeader: {
      text: s__(
        'DevopsAdoption|Feature adoption is based on usage in the last calendar month. Last updated: %{timestamp}.',
      ),
      button: s__('DevopsAdoption|Add new segment'),
    },
  },
  emptyState: {
    title: s__('DevopsAdoption|Add a segment to get started'),
    description: s__(
      'DevopsAdoption|DevOps adoption uses segments to track adoption across key features. Segments are a way to track multiple related projects and groups at once. For example, you could create a segment for the engineering department or a particular product team.',
    ),
    button: s__('DevopsAdoption|Add new segment'),
  },
  modal: {
    addingTitle: s__('DevopsAdoption|New segment'),
    editingTitle: s__('DevopsAdoption|Edit segment'),
    addingButton: s__('DevopsAdoption|Create new segment'),
    editingButton: s__('DevopsAdoption|Save changes'),
    cancel: __('Cancel'),
    namePlaceholder: s__('DevopsAdoption|My segment'),
    filterPlaceholder: s__('DevopsAdoption|Filter by name'),
    nameLabel: s__('DevopsAdoption|Name'),
    selectedGroupsTextSingular: s__('DevopsAdoption|%{selectedCount} group selected (20 max)'),
    selectedGroupsTextPlural: s__('DevopsAdoption|%{selectedCount} groups selected (20 max)'),
    error: s__('DevopsAdoption|An error occured while saving the segment. Please try again.'),
    noResults: s__('DevopsAdoption|No filter results.'),
  },
  table: {
    editButton: s__('DevopsAdoption|Edit segment'),
    deleteButton: s__('DevopsAdoption|Delete segment'),
    headers: {
      name: {
        label: s__('DevopsAdoption|Segment'),
      },
      issueOpened: {
        label: s__('DevopsAdoption|Issues'),
        tooltip: s__('DevopsAdoption|At least 1 issue opened'),
      },
      mergeRequestOpened: {
        label: s__('DevopsAdoption|MRs'),
        tooltip: s__('DevopsAdoption|At least 1 MR opened'),
      },
      mergeRequestApproved: {
        label: s__('DevopsAdoption|Approvals'),
        tooltip: s__('DevopsAdoption|At least 1 approval on an MR'),
      },
      runnerConfigured: {
        label: s__('DevopsAdoption|Runners'),
        tooltip: s__('DevopsAdoption|Runner configured for project/group'),
      },
      pipelineSucceeded: {
        label: s__('DevopsAdoption|Pipelines'),
        tooltip: s__('DevopsAdoption|At least 1 pipeline successfully run'),
      },
      deploySucceeded: {
        label: s__('DevopsAdoption|Deploys'),
        tooltip: s__('DevopsAdoption|At least 1 deploy'),
      },
      securityScanSucceeded: {
        label: s__('DevopsAdoption|Scanning'),
        tooltip: s__('DevopsAdoption|At least 1 security scan of any type run in pipeline'),
      },
    },
  },
  deleteModal: {
    title: s__('DevopsAdoption|Confirm delete segment'),
    confirmationMessage: s__('DevopsAdoption|Are you sure that you would like to delete %{name}?'),
    cancel: __('Cancel'),
    confirm: s__('DevopsAdoption|Delete segment'),
    error: s__('DevopsAdoption|An error occured while deleting the segment. Please try again.'),
  },
  tableCell: {
    trueText: s__('DevopsAdoption|Adopted'),
    falseText: s__('DevopsAdoption|Not adopted'),
  },
};

export const DEVOPS_ADOPTION_TABLE_TEST_IDS = {
  TABLE_HEADERS: 'header',
  SEGMENT: 'segmentCol',
  ISSUES: 'issuesCol',
  MRS: 'mrsCol',
  APPROVALS: 'approvalsCol',
  RUNNERS: 'runnersCol',
  PIPELINES: 'pipelinesCol',
  DEPLOYS: 'deploysCol',
  ACTIONS: 'actionsCol',
  SCANNING: 'scanningCol',
};
