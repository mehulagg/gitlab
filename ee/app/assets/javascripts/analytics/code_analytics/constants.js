export const PROJECTS_PER_PAGE = 50;
export const FILE_QUANTITIES = [25, 50, 100, 250, 500];
export const DEFAULT_FILE_QUANTITY = FILE_QUANTITIES[2];
export const DEFAULT_DAYS_IN_PAST = 30;
export const DEFAULT_SPACING = 5;
