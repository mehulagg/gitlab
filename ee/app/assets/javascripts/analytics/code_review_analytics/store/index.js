import Vue from 'vue';
import Vuex from 'vuex';
import filters from './modules/filters/index';
import mergeRequests from './modules/merge_requests/index';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    modules: {
      filters,
      mergeRequests,
    },
  });

export default createStore();
