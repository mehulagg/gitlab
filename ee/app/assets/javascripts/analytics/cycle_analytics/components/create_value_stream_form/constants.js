import { s__ } from '~/locale';

export const I18N = {
  SELECT_START_EVENT: s__('CustomCycleAnalytics|Select start event'),
  SELECT_END_EVENT: s__('CustomCycleAnalytics|Select stop event'),
};

export const ERRORS = {
  START_EVENT_REQUIRED: s__('CustomCycleAnalytics|Please select a start event first'),
  STAGE_NAME_EXISTS: s__('CustomCycleAnalytics|Stage name already exists'),
  INVALID_EVENT_PAIRS: s__(
    'CustomCycleAnalytics|Start event changed, please select a valid end event',
  ),
};

export const defaultErrors = {
  id: [],
  name: [],
  startEventIdentifier: [],
  startEventLabelId: [],
  endEventIdentifier: [],
  endEventLabelId: [],
};

export const defaultFields = {
  id: null,
  name: null,
  startEventIdentifier: null,
  startEventLabelId: null,
  endEventIdentifier: null,
  endEventLabelId: null,
};
