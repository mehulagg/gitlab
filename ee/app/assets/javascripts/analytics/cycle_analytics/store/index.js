import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';
import durationChart from './modules/duration_chart/index';
import typeOfWork from './modules/type_of_work/index';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    actions,
    getters,
    mutations,
    state,
    modules: { durationChart, typeOfWork },
  });
