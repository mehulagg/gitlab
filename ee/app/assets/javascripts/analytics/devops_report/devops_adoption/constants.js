import { s__, __, sprintf } from '~/locale';

export const DEFAULT_POLLING_INTERVAL = 30000;

export const MAX_SEGMENTS = 30;

export const MAX_REQUEST_COUNT = 10;

export const DEVOPS_ADOPTION_SEGMENT_MODAL_ID = 'devopsSegmentModal';

export const DEVOPS_ADOPTION_SEGMENT_DELETE_MODAL_ID = 'devopsSegmentDeleteModal';

export const DATE_TIME_FORMAT = 'yyyy-mm-dd HH:MM';

export const DEVOPS_ADOPTION_ERROR_KEYS = {
  groups: 'groupsError',
  segments: 'segmentsError',
  addSegment: 'addSegmentsError',
};

export const DEVOPS_ADOPTION_GROUP_LEVEL_LABEL = s__('DevopsAdoption|Add/remove sub-groups');

export const DEVOPS_ADOPTION_TABLE_REMOVE_BUTTON_DISABLED = s__(
  'DevopsAdoption|You cannot remove the group you are currently in.',
);

export const DEVOPS_ADOPTION_STRINGS = {
  app: {
    [DEVOPS_ADOPTION_ERROR_KEYS.groups]: s__(
      'DevopsAdoption|There was an error fetching Groups. Please refresh the page.',
    ),
    [DEVOPS_ADOPTION_ERROR_KEYS.segments]: s__(
      'DevopsAdoption|There was an error fetching Group adoption data. Please refresh the page.',
    ),
    [DEVOPS_ADOPTION_ERROR_KEYS.addSegment]: s__(
      'DevopsAdoption|There was an error enabling the current group. Please refresh the page.',
    ),
    tableHeader: {
      text: s__(
        'DevopsAdoption|Feature adoption is based on usage in the current calendar month. Last updated: %{timestamp}.',
      ),
      button: s__('DevopsAdoption|Add/remove groups'),
      buttonTooltip: sprintf(s__('DevopsAdoption|Maximum %{maxSegments} groups allowed'), {
        maxSegments: MAX_SEGMENTS,
      }),
    },
  },
  emptyState: {
    title: s__('DevopsAdoption|Add a group to get started'),
    description: s__(
      'DevopsAdoption|DevOps adoption tracks the use of key features across your favorite groups. Add a group to the table to begin.',
    ),
    button: s__('DevopsAdoption|Add Group'),
  },
  modal: {
    addingTitle: s__('DevopsAdoption|Add/remove groups'),
    addingButton: s__('DevopsAdoption|Save changes'),
    cancel: __('Cancel'),
    namePlaceholder: s__('DevopsAdoption|My group'),
    filterPlaceholder: s__('DevopsAdoption|Filter by name'),
    error: s__('DevopsAdoption|An error occurred while saving changes. Please try again.'),
    noResults: s__('DevopsAdoption|No filter results.'),
  },
  table: {
    removeButton: s__('DevopsAdoption|Remove Group from the table.'),
  },
  deleteModal: {
    title: s__('DevopsAdoption|Confirm remove Group'),
    confirmationMessage: s__(
      'DevopsAdoption|Are you sure that you would like to remove %{name} from the table?',
    ),
    cancel: __('Cancel'),
    confirm: s__('DevopsAdoption|Remove Group'),
    error: s__('DevopsAdoption|An error occurred while removing the group. Please try again.'),
  },
  tableCell: {
    trueText: s__('DevopsAdoption|Adopted'),
    falseText: s__('DevopsAdoption|Not adopted'),
  },
};

export const DEVOPS_ADOPTION_TABLE_TEST_IDS = {
  TABLE_HEADERS: 'header',
  SEGMENT: 'segmentCol',
  ISSUES: 'issuesCol',
  MRS: 'mrsCol',
  APPROVALS: 'approvalsCol',
  RUNNERS: 'runnersCol',
  PIPELINES: 'pipelinesCol',
  DEPLOYS: 'deploysCol',
  ACTIONS: 'actionsCol',
  SCANNING: 'scanningCol',
  LOCAL_STORAGE_SORT_BY: 'localStorageSortBy',
  LOCAL_STORAGE_SORT_DESC: 'localStorageSortDesc',
};

export const DEVOPS_ADOPTION_SEGMENTS_TABLE_SORT_BY_STORAGE_KEY =
  'devops_adoption_segments_table_sort_by';

export const DEVOPS_ADOPTION_SEGMENTS_TABLE_SORT_DESC_STORAGE_KEY =
  'devops_adoption_segments_table_sort_desc';

export const DEVOPS_ADOPTION_GROUP_COL_LABEL = s__('DevopsAdoption|Group');

export const DEVOPS_ADOPTION_TABLE_CONFIGURATION = [
  {
    title: s__('DevopsAdoption|Adoption'),
    cols: [
      {
        key: 'issueOpened',
        label: s__('DevopsAdoption|Issues'),
        tooltip: s__('DevopsAdoption|At least 1 issue opened'),
      },
      {
        key: 'mergeRequestOpened',
        label: s__('DevopsAdoption|MRs'),
        tooltip: s__('DevopsAdoption|At least 1 MR opened'),
      },
      {
        key: 'mergeRequestApproved',
        label: s__('DevopsAdoption|Approvals'),
        tooltip: s__('DevopsAdoption|At least 1 approval on an MR'),
      },
      {
        key: 'runnerConfigured',
        label: s__('DevopsAdoption|Runners'),
        tooltip: s__('DevopsAdoption|Runner configured for project/group'),
      },
      {
        key: 'pipelineSucceeded',
        label: s__('DevopsAdoption|Pipelines'),
        tooltip: s__('DevopsAdoption|At least 1 pipeline successfully run'),
      },
      {
        key: 'deploySucceeded',
        label: s__('DevopsAdoption|Deploys'),
        tooltip: s__('DevopsAdoption|At least 1 deploy'),
      },
      {
        key: 'securityScanSucceeded',
        label: s__('DevopsAdoption|Scanning'),
        tooltip: s__('DevopsAdoption|At least 1 security scan of any type run in pipeline'),
      },
    ],
  },
];
