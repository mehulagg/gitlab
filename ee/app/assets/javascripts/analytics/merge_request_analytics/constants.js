import { __ } from '~/locale';

export const DEFAULT_NUMBER_OF_DAYS = 365;
export const THROUGHPUT_STRINGS = {
  CHART_TITLE: __('Throughput'),
  Y_AXIS_TITLE: __('Merge Requests merged'),
  X_AXIS_TITLE: __('Month'),
  CHART_DESCRIPTION: __('The number of merge requests merged by month.'),
  NO_DATA: __('There is no data available.'),
};
