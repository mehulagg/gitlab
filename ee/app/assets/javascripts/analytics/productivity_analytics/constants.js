import { __, s__ } from '~/locale';

export const chartKeys = {
  main: 'main',
  timeBasedHistogram: 'timeBasedHistogram',
  commitBasedHistogram: 'commitBasedHistogram',
  scatterplot: 'scatterplot',
};

export const chartTypes = {
  histogram: 'histogram',
  scatterplot: 'scatterplot',
};

export const metricTypes = [
  {
    key: 'time_to_first_comment',
    label: __('Time from first commit until first comment'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_last_commit',
    label: __('Time from first comment to last commit'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_merge',
    label: __('Time from last commit to merge'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'commits_count',
    label: __('Number of commits per MR'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'loc_per_commit',
    label: __('Number of LOCs per commit'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'files_touched',
    label: __('Number of files touched'),
    chart: chartKeys.commitBasedHistogram,
  },
];

export const tableSortFields = metricTypes.reduce(
  (acc, curr) => {
    const { key, label, chart } = curr;
    if (chart === chartKeys.timeBasedHistogram) {
      acc[key] = label;
    }
    return acc;
  },
  { days_to_merge: __('Days to merge') },
);

export const tableSortOrder = {
  asc: {
    title: s__('ProductivityAnalytics|Ascending'),
    value: 'asc',
    icon: 'sort-lowest',
  },
  desc: {
    title: s__('ProductivityAnalytics|Descending'),
    value: 'desc',
    icon: 'sort-highest',
  },
};

export const timeToMergeMetric = 'time_to_merge';

export const defaultMaxColumnChartItemsPerPage = 20;
export const maxColumnChartItemsPerPage = {
  [chartKeys.main]: 40,
};

export const dataZoomOptions = [
  {
    type: 'slider',
    bottom: 10,
    start: 0,
  },
  {
    type: 'inside',
    start: 0,
  },
];

/**
 * #418cd8 --> $blue-400 (see variables.scss)
 */
export const columnHighlightStyle = { color: '#418cd8', opacity: 0.8 };

// The number of days which will be to the state's daysInPast
// This is required to query historical data from the API to draw a 30 days rolling median line
export const scatterPlotAddonQueryDays = 30;

export const accessLevelReporter = 20;
