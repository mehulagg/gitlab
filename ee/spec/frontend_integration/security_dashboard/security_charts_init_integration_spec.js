import { TEST_HOST } from 'jest/helpers/test_constants';
import { DASHBOARD_TYPES } from 'ee/security_dashboard/store/constants';
import initSecurityCharts from 'ee/security_dashboard/security_charts_init';

const TEST_DATASET = {
  link: '/test/link',
  svgPath: '/test/no_changes_state.svg',
  dashboardDocumentation: '/test/dashboard_page',
  emptyStateSvgPath: '/test/empty_state.svg',
};

describe('Security Charts', () => {
  let vm;
  let root;

  beforeEach(() => {
    root = document.createElement('div');
    document.body.appendChild(root);

    global.jsdom.reconfigure({
      url: `${TEST_HOST}/-/security/dashboard`,
    });
  });

  afterEach(() => {
    vm.$destroy();
    vm = null;
    root.remove();
  });

  const createComponent = ({ type, CUSTOM_DATA }) => {
    const el = document.createElement('div');
    Object.assign(el.dataset, { ...TEST_DATASET, ...CUSTOM_DATA });
    root.appendChild(el);
    vm = initSecurityCharts(el, type);
  };

  it('runs for the group-level', () => {
    createComponent({ type: DASHBOARD_TYPES.GROUP, CUSTOM_DATA: { groupFullPath: '/test/' } });

    expect(root).toMatchSnapshot();
  });

  it('runs for the instance-level', () => {
    createComponent({
      type: DASHBOARD_TYPES.INSTANCE,
      CUSTOM_DATA: { instanceDashboardSettingsPath: '/instance/settings_page' },
    });

    expect(root).toMatchSnapshot();
  });
});
