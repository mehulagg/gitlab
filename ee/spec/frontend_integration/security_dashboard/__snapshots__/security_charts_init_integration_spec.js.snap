// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security Charts runs for the group-level 1`] = `
<div>
  <div
    data-testid="security-charts-layout"
  >
    <h2>
      Security Dashboard
    </h2>
    <div
      class="gl-spinner-container gl-mt-6"
    >
      <span
        aria-label="Loading"
        class="align-text-bottom gl-spinner gl-spinner-orange gl-spinner-lg"
      />
    </div>
    <div
      class="security-charts gl-display-flex gl-flex-wrap"
    />
  </div>
</div>
`;

exports[`Security Charts runs for the instance-level 1`] = `
<div>
  <div
    data-testid="security-charts-layout"
  >
    <h2>
      Security Dashboard
    </h2>
    <div
      class="gl-spinner-container gl-mt-6"
    >
      <span
        aria-label="Loading"
        class="align-text-bottom gl-spinner gl-spinner-orange gl-spinner-lg"
      />
    </div>
    <div
      class="security-charts gl-display-flex gl-flex-wrap"
    />
  </div>
</div>
`;
